cnd/src/btsieve/bitcoin/bitcoind_connector.rs

```diff
@@ -16,14 +18,17 @@ pub struct BitcoindConnector {
     chaininfo_url: Url,
     raw_block_by_hash_url: Url,
     client: Client,
+    block_cache: HashMap<sha256d::Hash, bitcoin::Block>,
 }
 
 impl BitcoindConnector {
     pub fn new(base_url: Url, _network: Network) -> Result<Self, reqwest::UrlError> {
+        let block_cache: HashMap<sha256d::Hash, bitcoin::Block> = HashMap::new();
```

Comment:
I'd rather have caching concern be solved orthogonally to this functionality so that we can reuse it for different kinds of connectors.

Reply:
Oh boy, I thought I was flying yesterday - seems I was swimming and I didn't know it.  And now in normal English; damn this is a mistake by me while separating work into distinct PRs, clearly refactor should not introduce new logic.  Thanks for the review, sorry to waste your time with silly mistakes.

Reply:
All good, don't be too so harsh on yourself :)
That is what we have our code review process for ;)